bin/test.py

```diff
@@ -231,16 +298,17 @@ def remove_keypairs(session, config, key_pairs):
     """ Remove key-pairs created as part of test harness
     """
 
-    for key_pair in key_pairs:
+    for region in get_regions(session.client('ec2', region_name='us-east-1', config=config)):
 
-        ec2_client = session.client('ec2', region_name=key_pair['Region'], config=config)
-        delete_key_pair(ec2_client, key_pair['KeyName'])
-        print 'Deleted keypair: {} in region: {}'.format(key_pair['KeyName'],key_pair['Region'])
+        ec2_client = session.client('ec2', region_name=region, config=config)
+        delete_key_pair(ec2_client, key_pairs[region]['KeyName'])
+        print 'Deleted keypair: {} in region: {}'.format(key_pairs[region]['KeyName'],region)
 
-def test_stack(session, config, stack_type):
+def test_stack(session, config, stack_type, username, repository, token, key_pairs):
```

Comment:
As the parameters are growing, we should wrap session, aws_config, username, repository, token and key_pairs into an object/dictionary.